### PR TITLE
Feature/#7 search

### LINKED ISSUE #7: Add search
look for a specific test case

https://github.com/maxbeatty/jsperf.com/blob/breakdown/_tpl/search.tpl

### PR DESCRIPTION
**Pull request by MadaraUchiha**
Aims to close #7.

This is an ongoing PR. I will add tests and stuff to it tomorrow.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -29,6 +29,7 @@
   "dependencies": {
     "bell": "^3.0.0",
     "blipp": "^2.0.1",
+    "bluebird": "^2.9.30",
```
Comment (asilluron) [inline package.json]: Since we are using iojs, can't we can use built-in ES6 promises.

Comment (asilluron) [inline package.json]: I just saw what you are using bluebird for (promisifying), makes sense so far.

Comment (MadaraUchiha) [inline package.json]: Also, for most mainstream cases, Bluebird Promises are faster than native V8 Promises. Pref all the way! :dash:

**Comment by MadaraUchiha** [comment]
This PR is ready from my end.

Although, I still can't get tests to pass, and my ESLint is complaining about invalid eslintrc file.

File: server/web/search/index.hbs [hunk 2]
```diff
@@ -0,0 +1,27 @@
+<h1>Search results</h1>
+
+{{#if result}}
+<ul>
+  {{#each result}}
+    <!--<li>
```
Comment (maxbeatty) [inline server/web/search/index.hbs]: the old PHP is safe to remove

**Comment by maxbeatty** [comment]
I, too, noticed my eslint plugin for atom was complaining. I'll try to address that in a separate branch.

Looks like your test coverage isn't 100%

> Coverage: 99.41% (8/1358)
server/repositories/pages.js missing coverage on line(s): 148, 155, 156, 157
server/services/pages.js missing coverage on line(s): 76
server/web/search/index.js missing coverage on line(s): 11, 12, 14
Code coverage below threshold: 99.41 < 100

You can use `npm run test-cov` and `npm run test-cov-html` to help bring it up

**Comment by MadaraUchiha** [comment]
@maxbeatty You are correct! I've added the missing tests.

**Comment by maxbeatty** [comment]
```
$ npm run test-cov

> jsperf.com@2.0.0 test-cov /Users/max/Sites/so/jsperf.com/jsperf.com
> lab -t 100 -c -L


  
  ..................................................
  ..................................................
  ........................

124 tests complete
Test duration: 2619 ms
No global variable leaks detected
Coverage: 99.63% (5/1358)
server/repositories/pages.js missing coverage on line(s): 148, 155, 156, 157
server/services/pages.js missing coverage on line(s): 76
Code coverage below threshold: 99.63 < 100
Linting results:
	test/server/web/search/index.js:
		Line 11: quotes - Strings must use doublequote.
```

**Comment by MadaraUchiha** [comment]
Can you get in IRC? I'm obviously missing something here.

**Comment by maxbeatty** [comment]
@MadaraUchiha thanks for doing the bulk of this work! I pulled in your changes along with a few more tests in https://github.com/jsperf/jsperf.com/commit/0715e7f5e7814d4b3db93eedcfded187f7c40ae7

All of your tests were passing but you didn't have 100% test coverage and had a linting error.

> Coverage: 99.63% (5/1358)
server/repositories/pages.js missing coverage on line(s): 148, 155, 156, 157
server/services/pages.js missing coverage on line(s): 76

If you run `npm run test-cov-html`, an HTML file will be generated and opened in your default browser showing where coverage is missing. In this case, you needed to add a test for the page service and repository. Hope this helps get everything green next time!

**Comment by MadaraUchiha** [comment]
@maxbeatty I consistently get about 95% of test coverage. None of the GitHub integration tests appear to be there, or aren't covering the code. So even when I did reach 100% coverage on my part, I had no actual way of getting a passing build.

**Comment by asilluron** [comment]
Sounds like you need to rebase to make sure you have the latest on master

**Pull request closed without merging**